core/lib/eth_client/src/clients/http/query.rs

```diff
@@ -361,6 +361,11 @@ where
                 .with_arg("chunk_end", &chunk_end));
         }
 
+        if fee_history.base_fee_per_gas.len() != fee_history.base_fee_per_blob_gas.len() {
```

Comment:
Can you move this section to line 338?

Otherwise this line may not pass using gcp json rpc provider 

```
if fee_history.base_fee_per_blob_gas.len() != chunk_size + 1 {
```

Reply:
done!